### PR TITLE
implements authentication in the frontend

### LINKED ISSUE #18: implement frontend authentication
This issue includes the following points:

- [x] Connect register and login form to backend
- [x] Store tokens in local storage
- [x] Request interceptor for attaching token to requests
- [x] Request interceptor for refreshing auth-token on 401 response

### PR DESCRIPTION
**Pull request by PaulKalho**
closes #18 

we need to improve the error handling (see #28)

merge #32 before!

### CODE REVIEW

**Comment by PaulKalho** [comment]
Tested and works. R4R @mottegk @use-to

File: docker-compose.yml [hunk 1]
```diff
@@ -228,16 +228,18 @@ services:
   frontend:
     build: frontend
     restart: always
+    environment:
+      NEXT_PUBLIC_API_URL: "http://localhost:4000"
     ports:
-      - "8000:80"
+      - "8008:80"
```
Comment (use-to) [inline docker-compose.yml:234]: Unnecessary change

File: docker-compose.yml [hunk 2]
```diff
@@ -228,16 +228,18 @@ services:
...
   
   core:
     build: core
     restart: always
     ports:
       - "4000:4000"
     environment:
-      EXTERNAL_URL: "http://localhost:4000"
+      EXTERNAL_URL: "http://localhost:8008"
```
Comment (use-to) [inline docker-compose.yml:242]: See above

File: frontend/mutations/userMutation.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+function useRegisterMutation() {
+  return useMutation({
+    mutationFn: async function register(user: UserDTO) {
+      try {
+        const response = await api.post(REGISTER_ENDPOINT, JSON.stringify(user))
+
+        return response.data
+      } catch (error) {
```
Comment (use-to) [inline frontend/mutations/userMutation.ts:29]: Use react-query's error handling instead.

**Review by use-to — CHANGES_REQUESTED** [review]
Good pull request providing important traction. A few requests as this is a pivotal change.

**Review by use-to — APPROVED** [review]

**Pull request merged**